scripts/lisdf_plan_to_reset.py

```diff
@@ -0,0 +1,105 @@
+               plan.commands))
+    assert len(path_commands) > 0
+    first_path_command = cast(JointSpacePath, path_commands[0])
+    return first_path_command.waypoint(0)
+
+
+def _main(lisdf_filepath: Path, output_filepath: Path, seed: int = 0) -> None:
+    utils.reset_config({"seed": seed})
+    # Load the LISDF plan.
+    with open(lisdf_filepath, "r", encoding="utf-8") as f:
+        plan_json = f.read()
+        plan = cast(LISDFPlan, LISDFPlan.from_json(plan_json))
+    # Set up PyBullet.
+    physics_client_id = p.connect(p.DIRECT)
+    table_pose = PyBulletBlocksEnv._table_pose  # pylint: disable=protected-access
+    table_orientation = PyBulletBlocksEnv._table_orientation  # pylint: disable=protected-access
+    table_id = p.loadURDF(utils.get_env_asset_path("urdf/table.urdf"),
+                          useFixedBase=True,
+                          physicsClientId=physics_client_id)
+    p.resetBasePositionAndOrientation(table_id,
+                                      table_pose,
+                                      table_orientation,
+                                      physicsClientId=physics_client_id)
+    ee_home = (PyBulletBlocksEnv.robot_init_x, PyBulletBlocksEnv.robot_init_y,
+               PyBulletBlocksEnv.robot_init_z)
+    robot = create_single_arm_pybullet_robot("panda", physics_client_id,
+                                             ee_home)
+    # Get the initial robot state in the plan.
+    gripper_idxs = [robot.left_finger_joint_idx, robot.right_finger_joint_idx]
+    joint_names = robot.arm_joint_names
+    arm_joint_names = list(np.delete(joint_names, gripper_idxs))
+    plan_init_joints = _get_first_joints_from_lisdf(plan)
+    # Set up the Panda client.
+    client = PandaClient()
+    # Get the current robot state.
+    current_joints = client.get_joint_positions()
+    # Create a motion plan from the current state to the first state in the
+    # LISDF plan. For now, ignore possible collisions with other objects.
+    # Assume grippers are open, as that makes motion planning more difficult
+    # and hence safer.
+    current_joints[robot.left_finger_joint_name] = robot.open_fingers
+    current_joints[robot.right_finger_joint_name] = robot.open_fingers
+    plan_init_joints[robot.left_finger_joint_name] = robot.open_fingers
+    plan_init_joints[robot.right_finger_joint_name] = robot.open_fingers
+    current_joint_lst = [
+        current_joints[joint_name] for joint_name in joint_names
+    ]
+    plan_init_joint_lst = [
+        plan_init_joints[joint_name] for joint_name in joint_names
+    ]
+    motion_plan = run_motion_planning(robot,
+                                      current_joint_lst,
+                                      plan_init_joint_lst,
+                                      collision_bodies={table_id},
+                                      seed=seed,
+                                      physics_client_id=physics_client_id)
+    assert motion_plan is not None
```

Comment:
Could `motion_plan == []`? Would that happen when the robot is already at the start position (usually it would never exactly be at that start position but close I'd think as you can't get the joint positions to that level of precision).

In that case, we can just write the existing plan I guess.

Reply:
yeah I think if it's empty then everything should work